src/src/NumberRange.java

```diff
@@ -0,0 +1,129 @@
+/**
```

Comment:
There is just way too much comments in your code. Here are some reasons this is a bad thing:
- distracting
- most are useless. Trust that you are writing this code for other programmers. Is there really a need to comment on a method called "join" (and the comment is - "join to number ranges")?
* maintenance nightmare. All code changes in time. Comments rarely do. So in time, these comments will be wrong as the code changes.
* We have unit tests to document our code, making comments even less needed.

Be minimalistic with your comments. Only do so when absolutely necessary.